backend/services/implementations/testSessionService.ts

```diff
@@ -213,6 +212,51 @@ class TestSessionService implements ITestSessionService {
     return testSessionDtos;
   }
 
+  async updateTestSession(
+    id: string,
+    testSession: TestSessionRequestDTO,
+  ): Promise<TestSessionResponseDTO> {
+    let updatedTestSession: TestSession | null;
+
+    try {
+      const { results } = testSession;
+      if (results) {
+        await Promise.all(
+          results.map(async (result: ResultRequestDTO, i) => {
+            if (result.gradingStatus === GradingStatus.UNGRADED) {
+              const gradedResult: ResultResponseDTO = await this.gradeTestResult(
+                result,
+                id,
+              );
+              results[i] = gradedResult;
+            }
+          }),
+        );
+      }
+
+      updatedTestSession = await MgTestSession.findByIdAndUpdate(
+        id,
+        testSession,
+        {
+          new: true,
+          runValidators: true,
```

Comment:
does this enable automated type-checking?

Reply:
yees `runValidators=true` will validate the update operation against the model's schema

and `new=true` returns the modified document rather than the original